### PR TITLE
Rename yaml parameters

### PR DESCRIPTION
**Pull request by lucekdudek**
- Renamed `budget` to `budget_limit`
 - Renamed `cost_management` to `budget_control`
 - Renamed `budget_control` parameters

### CODE REVIEW

**Review by mateuszsrebrny — APPROVED** [review]
lgtm, besides question about the line in network-stats output
```
Passed Reject if max_average_usage_cost exceeds 1.5: 48
```

It might be generated automatically - I just don't see it here in the diff :)

File: golem-cluster.yaml [hunk 1]
```diff
@@ -35,16 +35,16 @@ provider:
...
+      budget_control:
+        # Estimated average load and duration for worker that tells budget control to pick the least expensive Golem provider offers first.
         # If not provided, offers will be picked at random.
         # Both values need to be defined or undefined together.
-        average_cpu_load: 0.8
-        average_duration_minutes: 20
+        expected_cpu_load: 0.8
+        expected_duration_minutes: 20
 
         # Amount of GLMs for average usage which Golem provider offer will be rejected if exceeded.
```
Comment (mateuszsrebrny) [inline golem-cluster.yaml:45]:
```suggestion
        # Amount of GLMs for expected usage which Golem provider offer will be rejected if exceeded.
```

**Comment by lucekdudek** [comment]
> lgtm, besides question about the line in network-stats output
> 
> ```
> Passed Reject if max_average_usage_cost exceeds 1.5: 48
> ```
> 
> It might be generated automatically - I just don't see it here in the diff :)

It is generated automatically

File: ray_on_golem/server/services/golem/golem.py [hunk 2]
```diff
@@ -104,18 +104,20 @@ def _get_hash_from_node_config(node_config: NodeConfigData) -> str:
...
     ) -> ManagerStack:
         stack = ManagerStack()
 
         payload = await self._demand_config_helper.get_payload_from_demand_config(
             node_config.demand
         )
 
-        ManagerStackNodeConfigHelper.apply_cost_management_avg_usage(stack, node_config)
-        ManagerStackNodeConfigHelper.apply_cost_management_hard_limits(stack, node_config)
+        ManagerStackNodeConfigHelper.apply_budget_control_avg_usage(stack, node_config)
```
Comment (approxit) [inline ray_on_golem/server/services/golem/golem.py:115]:
```suggestion
        ManagerStackNodeConfigHelper.apply_budget_control_expected_usage(stack, node_config)
```

File: ray_on_golem/network_stats/services/network_stats.py [hunk 3]
```diff
@@ -178,18 +178,20 @@ async def _consume_draft_proposals(self, stack: ManagerStack) -> None:
...
     ) -> ManagerStack:
         stack = ManagerStack()
 
         payload = await self._demand_config_helper.get_payload_from_demand_config(
             node_config.demand
         )
 
-        ManagerStackNodeConfigHelper.apply_cost_management_avg_usage(stack, node_config)
-        ManagerStackNodeConfigHelper.apply_cost_management_hard_limits(stack, node_config)
+        ManagerStackNodeConfigHelper.apply_budget_control_avg_usage(stack, node_config)
```
Comment (approxit) [inline ray_on_golem/network_stats/services/network_stats.py:189]:
```suggestion
        ManagerStackNodeConfigHelper.apply_budget_control_expected_usage(stack, node_config)
```

**Review by approxit — APPROVED** [review]
With minor gotchas.

**Pull request merged**